build/helper/metadata_add_all.py

```diff
@@ -441,11 +442,17 @@ def add_all_attribute_metadata(attributes, config):
 def _add_enum_codegen_method(enums, config):
     '''Adds 'codegen_method' that will determine whether and how the enum is code genned. Default is public
 
-    Set all to 'no', then go through all functions and attributes and set to least restrictive use
+    Set all to 'no', then go through all functions and attributes and set to least restrictive use.
+    If the 'codegen_method' of an enum has been explicitly set to a value other than 'public' and it
+    is used by a 'public'/'python-only' function or 'public' attribute, a ValueError would be thrown
     '''
+    enum_to_has_explicit_codegen_method = {}
```

Comment:
My initial rationale of naming it as such was to indicate that it's a dict using a `<key>_to_<value>` naming convention, since it's a dict that maps an `enum` instance to `has_explicit_codegen_method` boolean.

Nevertheless, I agree that this name is awkward and I have renamed it to `enum_has_explicit_codegen_method` in my latest commit.